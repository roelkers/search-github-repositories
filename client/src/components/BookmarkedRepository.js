import React from 'react';

export default ({ repository: { id, name, owner, forks, stars }, onDeleteBookmark }) => {
  return (
    <div className='repository'>
      <h2>{name}</h2>
      <p>Id: {id}</p>
      <p>Owner: {owner}</p>
      <p>Forks: {forks}</p>
      <p>Stars: {stars}</p>
      <button type="button" onClick={() => onDeleteBookmark(id)}>
        Remove Bookmark
      </button>
    </div>
  );
};