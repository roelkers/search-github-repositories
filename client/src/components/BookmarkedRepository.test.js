import React from 'react';
import BookmarkedRepository from './BookmarkedRepository';
import renderer from 'react-test-renderer';

describe('BookmarkedRepository', () => {
    const repository = {
        id : 1,
        name : "test",
        owner : "test_owner",
        stars : 2,
        forks : 3
    };
    it('should match the snapshot', () => {
        const component = renderer.create(<BookmarkedRepository repository={repository} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});