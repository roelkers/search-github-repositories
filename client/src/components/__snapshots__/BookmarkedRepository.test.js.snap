// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkedRepository should match the snapshot 1`] = `
<div
  className="repository"
>
  <h2>
    test
  </h2>
  <p>
    Id: 
    1
  </p>
  <p>
    Owner: 
    test_owner
  </p>
  <p>
    Forks: 
    3
  </p>
  <p>
    Stars: 
    2
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Remove Bookmark
  </button>
</div>
`;
